### PR TITLE
feat: comment section ui improvement

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- The `MainComment` and `SubComment` was pretty tough to maintain as we always have to touch two components whenever we need to make changes to the comment component. To make it easier, I created a generic component that simply fits both scenarios in terms of functionality.

Quick link: https://preview.app.daily.dev/posts/JrnJMPYqG#c-I2mojWywK

Local previews:
Screenshot 2023-01-11 at 9 06 21 PM
Screenshot 2023-01-11 at 9 05 34 PM
Screenshot 2023-01-11 at 8 49 31 PM
Screenshot 2023-01-11 at 8 49 20 PM

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done
Screenshot 2023-01-11 at 8 49 20 PM

### CODE REVIEW

**Comment by sshanzel** [comment]
@rebelchris @el feel free to review, the tests are just failing (different call points) but code-wise and functionality, this should be ready for review.

File: packages/shared/src/components/comments/CommentBox.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            <CommentAuthor
+              postAuthorId={postAuthorId}
+              author={comment.author}
+              appendTooltipTo={appendTooltipTo}
+            />
+            {comment.author?.id === postScoutId && <ScoutBadge />}
+          </FlexRow>
+          <FlexRow className="items-center text-theme-label-quaternary">
+            <ProfileLink user={comment.author}>
+              @{comment.author?.username}
```
Comment (rebelchris) [inline packages/shared/src/components/comments/CommentBox.tsx:87]: What case would we not have the author? (It would show `@` then is that intended?

**Review by rebelchris — APPROVED** [review]
Nice, this is amazing!
Good refactor on making it more generic.

Please just double check:
- Extension
- Mobile
- Companion!
- The empty comments 
- The one comment margin bottom (we have this issue now)

File: packages/shared/src/components/comments/CommentBox.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/comments/CommentBox.tsx:87]: Oh, we can remove this optional chaining. Should never happen

**Review by el — CHANGES_REQUESTED** [review]
There seems to be an issue with the preview after comment edit. (might not be introduced here not sure)

Can you fix this before we finalize it?


https://user-images.githubusercontent.com/1488164/212012946-81a3a33f-d244-4e03-9ba9-3e9cc04893b5.mov

**Comment by sshanzel** [comment]
@el I think it was intentional before to disable preview since there was no change and that is already what you see before clicking the edit button. But if you think we should change this, then I will remove this condition.
image

**Comment by rebelchris** [comment]
@sshanzel Noticed the "Post" button is not 100% conform new design would you mind checking:

Design:
image

**Comment by sshanzel** [comment]
> @sshanzel Noticed the "Post" button is not 100% conform new design would you mind checking:

From the Sketch though it says a Secondary button and the element is having a `btn-secondary`.
image

**Comment by sshanzel** [comment]
@el let me know if the issue still persists, I have pushed a fix and tried it on my end. Seems to work as intended now. Thank you for the review!

**Review by el — APPROVED** [review]
Great work :shipit:

**Pull request merged**